src/backend/columnar/sql/columnar--10.2-2--10.2-3.sql

```diff
@@ -10,6 +10,17 @@
 --
 -- To do that, drop stripe_first_row_number_idx and create a unique
 -- constraint with the same name to keep the code change at minimum.
+--
+-- If we have a pg_depend entry for this index, we can not drop it as
+-- the extension depends on it. Remove the pg_depend entry if it exists.
```

Comment:
I am a bit baffled that the extension depends on an index instead of the index having an extension dependency to the extension preventing it from being dropped outside of `DROP EXTENSION`.

Could you explain it a bit better why a columnar catalog table has an index depending on the extension?